tests/datatypes/test_str.py

```diff
@@ -10,6 +10,30 @@ def test_setattr(self):
             print('Done.')
             """)
 
+    def test_endswith(self):
+        self.assertCodeExecution("""
+            s = "abracadabra"
+            suffix = "abra"
+            print(s.endswith(end))            
```

Comment:
I don't think these tests are doing what you think they're doing.

The variable end isn't declared and suffix is never used, I think they're all throwing NameError.